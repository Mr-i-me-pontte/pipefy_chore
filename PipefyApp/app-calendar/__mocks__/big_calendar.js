export const label = 'Testing label';
export const onNavigate = jest.fn();
export const onViewChange = jest.fn();
export const view = 'month';
export const views = ['month', 'week', 'day', 'agenda'];

export default { label, onNavigate, onViewChange, view, views };
