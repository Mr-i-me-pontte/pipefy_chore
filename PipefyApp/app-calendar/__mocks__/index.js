import MockDate from 'mockdate';

MockDate.set('2017-12-01');

export const card = {
  due_date: '2017-12-15T16:00:00-02:00',
  id: 'aBc_1234',
  internalId: '1',
  isVisible: true,
  title: 'Testing card',
};

export const e = { preventDefault: jest.fn() };

export const event = {
  end: new Date(new Date(card.due_date).getTime() + 30 * 60000),
  id: card.id,
  start: new Date(card.due_date),
  title: card.title,
};

export const pipefy = {
  app: { pipeId: '1' },
  locale: 'en',
  openCard: jest.fn(),
  allCards: () => Promise.resolve({ cards: [card], pageInfo: { hasNextPage: false } })
};
