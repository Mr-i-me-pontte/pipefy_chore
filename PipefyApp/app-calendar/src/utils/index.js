// @flow

import moment from 'moment';

import type { Card } from '../models';

const unit = view => (view === 'agenda' ? 'day' : view);

export const endDateByView = (date: Date, view: string) => {
  const processedDate = moment.utc(date);

  if (view === 'agenda') processedDate.add(30, 'days');

  return processedDate.endOf(unit(view)).toISOString();
};

export const startDateByView = (date: Date, view: string) =>
  moment
    .utc(date)
    .startOf(unit(view))
    .toISOString();

export const transformCardsToEvents = (cards: Array<Card>) =>
  cards.map(card => {
    const start = new Date(card.due_date);
    const end = new Date(start.getTime() + 30 * 60000);

    return {
      end,
      id: card.id,
      start,
      title: card.title,
    };
  });
