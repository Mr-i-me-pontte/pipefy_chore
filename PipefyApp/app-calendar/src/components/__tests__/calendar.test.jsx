import React from 'react';
import BigCalendar from 'react-big-calendar';
import { shallow } from 'enzyme';

import Calendar from '../calendar';
import { pipefy } from '../../../__mocks__';
import { flushPromises } from '../../utils/testUtils'

describe('Calendar', () => {
  it('render with big calendar', async () => {
    const wrapper = shallow(<Calendar pipefy={pipefy} />);
    expect(wrapper.find(BigCalendar)).toHaveLength(1);
    await flushPromises();
    wrapper.update();
    expect(wrapper.find(BigCalendar).props()).toMatchSnapshot();
  });
});
