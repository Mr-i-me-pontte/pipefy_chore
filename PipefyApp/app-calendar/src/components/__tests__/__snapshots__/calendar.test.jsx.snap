// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar render with big calendar 1`] = `
Object {
  "components": Object {
    "agenda": Object {
      "event": [Function],
    },
    "toolbar": [Function],
  },
  "culture": "en",
  "defaultDate": 2017-12-01T00:00:00.000Z,
  "defaultView": "month",
  "events": Array [
    Object {
      "end": 2017-12-15T18:30:00.000Z,
      "id": "aBc_1234",
      "start": 2017-12-15T18:00:00.000Z,
      "title": "Testing card",
    },
  ],
  "onNavigate": [Function],
  "onSelectEvent": [Function],
  "onView": [Function],
  "selectable": true,
}
`;
